### PR TITLE
beacon/blsync: remove cli dependencies

### PR DESCRIPTION
**Pull request by zsfelfoldi**
This PR moves chain config related code (config file processing, fork logic, network defaults) from `beacon/types` and `beacon/blsync` into `beacon/params` while the command line flag logic of the chain config is moved into `cmd/utils`, thereby removing the cli dependencies from package `beacon` and its sub-packages.

### CODE REVIEW

File: beacon/params/config.go [hunk 1]
```diff
@@ -39,88 +39,21 @@ const syncCommitteeDomain = 7
...
-
-func (f Forks) Len() int      { return len(f) }
-func (f Forks) Swap(i, j int) { f[i], f[j] = f[j], f[i] }
-func (f Forks) Less(i, j int) bool {
-	if f[i].Epoch != f[j].Epoch {
-		return f[i].Epoch < f[j].Epoch
-	}
-	return f[i].knownIndex < f[j].knownIndex
+// ClientConfig contains beacon light client configuration.
+type ClientConfig struct {
```
Comment (MariusVanDerWijden) [inline beacon/params/config.go:43]: Maybe `LightClientConfig` or `BeaconClientConfig`?
Just `ClientConfig` is a bit ambiguous in my opinion

Comment (MariusVanDerWijden) [reply beacon/params/config.go:43]: Ah nvm we're still in `beacon/params`. I thought you moved it into `/params`

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -1889,6 +1890,79 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
...
+	case ctx.Bool(MainnetFlag.Name):
+		config.ChainConfig = *bparams.MainnetLightConfig
+	case ctx.Bool(SepoliaFlag.Name):
+		config.ChainConfig = *bparams.SepoliaLightConfig
+	case ctx.Bool(HoleskyFlag.Name):
+		config.ChainConfig = *bparams.HoleskyLightConfig
+	default:
+		if !customConfig {
+			config.ChainConfig = *bparams.MainnetLightConfig
+		}
```
Comment (MariusVanDerWijden) [inline cmd/utils/flags.go:1909]: I think we should log here, that a fully custom config is being used

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM

File: cmd/utils/flags.go (hunk 2 above)
Comment (zsfelfoldi) [reply cmd/utils/flags.go:1909]: done

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**